### DISCUSSION TITLE
what does this eror means?

### DISCUSSION THREAD

**Discussion by aniel300** [euzu/tuliprox#140] [General]
`[2024-12-25T11:00:07Z DEBUG m3u_filter::utils::request_utils] Request took:0 secs,  Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("xxx.live")), port: Some(8443), path: "/player_api.php", query: Some("username=xxx&password=xxx&action=get_series_categories"), fragment: None }`

**Comment by euzu** [comment]
Only debug message.
--

**Comment by aniel300** [comment]
i also get this as well
`[2024-12-25T11:05:59Z DEBUG m3u_filter::utils::request_utils] Request took:0 secs,  Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("xxx.org")), port: Some(8080), path: "/player_api.php", query: Some("username=xxx&password=xxx&action=get_vod_categories"), fragment: None }`

**Comment by euzu** [comment]
Wait this are not  error messages, it is debug.
It says the  request to this url took 0 seconds (was fast, without delay)

**Reply by aniel300** [reply]
umm ok. can i add `''` to the user name and pass?. could it something related to invalid characters?
![Code_a2TREo49l4](https://github.com/user-attachments/assets/8c740979-402d-4c20-917b-4850bdc93917)

**Reply by euzu** [reply]
First at all,  your input/source is what ? m3u or xtream codes?

If it is Xtream-Codes you need user password.

If it is m3u you dont need. The credentials are in the url.

**Reply by aniel300** [reply]
it is a xc imput

**Reply by euzu** [reply]
If it is Xtream-Codes you need user password.

This has nothing to do with me, it is the xc api.

**Reply by aniel300** [reply]
yes that i know am asking if `''` would be needed around it since visual code is marking it in a strange way. also is there a way to include the update process within the info debug and not the debug debug envl, or a way to make it easier to know when it finished updating?

**Reply by euzu** [reply]
If this is a config file for m3u-filter, it uses yaml file format.
In yaml you can define a string like this:

```yaml
password: 'my very secret password'
```

**Reply by aniel300** [reply]
also is there a way to include the update process within the info debug and not the debug debug env, or a way to make it easier to know when it finished updating?

**Reply by aniel300** [reply]
> also is there a way to include the update process within the info debug and not the debug debug env, or a way to make it easier to know when it finished updating?

can u implement this? btw merry Christmas and I hope u feeling better

**Comment by euzu** [comment]
In the logs you can see a info output

```
2024-12-25T11:44:23Z INFO  m3u_filter::processing::playlist_processor] {"stats": {"name":"xtream","type":"xtream","errors":0,"raw":{"groups":23,"channels":1617},"processed":{"groups":3,"channels":4},"took":"1 secs"}}
```
This means  the update process is finished

**Reply by aniel300** [reply]
yes but because am using redirect, there is to many logs so most of the time i miss it. this is with logs set to debug, if u can set this update process the the default logs of `info` then i think this would help. if it is to much work then is ok as this is not the end of the world

**Reply by euzu** [reply]
This log is INFO level `2024-12-25T11:44:23Z INFO  m3u_filter::processing::playlist_processor] {"stats": {`

The default log level is INFO,  i think you enabled DEBUG log.

**Reply by aniel300** [reply]
is there a way to know when it finishes the update process when using log level of `info`?

**Reply by aniel300** [reply]
> This log is INFO level `2024-12-25T11:44:23Z INFO m3u_filter::processing::playlist_processor] {"stats": {`

when i see this in the `info` logs level is when the process has finished?

**Reply by euzu** [reply]
Yes

**Reply by aniel300** [reply]
ahh ok, thank u.

**Reply by euzu** [reply]
I have added a new output log  (INFO level)
`Update process finished!`
which will be in the next release.

**Reply by aniel300** [reply]
sweet. perfect bud.

**Reply by aniel300** [reply]
> I have added a new output log (INFO level) `Update process finished!` which will be in the next release.

can u add a way to know which m3u or xc are dead, fail, etc similar to how it is shown in the debug level upon the completion of sync on the info level one? trying to find that info when using the debug level while using redirect and heavy traffic/ request is a little hard if am not mistaken. i mainly trying to find out which m3u/xc are dead/bad to take them out.

**Comment by euzu** [comment]
yes i added a log at info level, will be in the next beta image.

**Reply by aniel300** [reply]
lets go

**Reply by aniel300** [reply]
was this added to `Release v2.1.1` or should i wait for the next beta?
